### ISSUE TITLE
config rpc api should reject unsupported http verbs

### ISSUE THREAD

**Issue by juliantaylor** [fluent/fluentd#4292]
### Describe the bug

The config rpc api (https://docs.fluentd.org/deployment/rpc) will hang indefinitely when you send a HTTP POST request to it. It should probably reject the unintended verbs or at least not hang.

We had an issue with our very simple config reloader not reloading after the first change:

```
inotifywait -mrq -e create -e modify -e delete /etc/fluentd-config | while read FILE
do
  curl -sS --fail -XPOST http://localhost:24444/api/config.reload
done
```

This curl request hangs indefinitely (as it is missing a timeout via the -m flag). using GET works fine and returns `{"ok":true}`

This may be a minor regression as this was running for 5+ years and we only noticed recently :) but maybe we just never saw as our fluentd pod lifetime and change rate is usually low

If this is intended the documentation should state how the behaviour depends on the verb used.

### To Reproduce

    curl -sS --fail -XPOST http://localhost:24444/api/config.reload

### Expected behavior

returns some http status

### Your Environment

```markdown
- Fluentd version: 1.15.3
- TD Agent version: 4.4.2-1
- Operating system:
- Kernel version:
```

### Your Configuration

```apache
<system>
    rpc_endpoint "127.0.0.1:24444"
  </system>

```
```


### Your Error Log

```shell
none
```

**Comment by daipom** [comment]
Thanks for your report.

POST without body causes hang.

This is caused by 

* https://github.com/ruby/webrick/issues/67

This is fixed in 

* https://github.com/ruby/webrick/commit/f87aec09fce09c142d2aa5108987338114327e4c

We need to use `Webrick` v1.8.0 or later.

I will check the current Fluend dependencies later.

**Comment by daipom** [comment]
This is fixed in td-agent v4.5.0 or later.
Please try it.
